etl-driver/src/main/java/com/wipro/ats/bdre/im/etl/api/sftp/SFTP.java

```diff
@@ -149,7 +149,8 @@ private void transferFile(Session session, String remoteFile, String localFile)
                     fos = new FileOutputStream(prefix == null ? localFile : prefix + file);
                     int foo;
                     while (true) {
-                        if (buf.length < filesize) foo = buf.length;
+                        if (buf.length < filesize)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)